### PR TITLE
Merged with base and a lot of improvements

### PR DESCRIPTION
**Pull request by friscoMad**
## Description
Too much going on:

* First of all merged with develop
* Added location to raid WH
* Made raid parsing dependant on gym parsing (if makes 0 sense to store raids with no-gym, as they wouldn't be displayed, but can revert if people wants raid only maps)
* Changed column name and wh attributes to be more descriptive
* Added raid info to gym sidebar
* Improved raid UI a little
* Fixed some issues with markers
* Changed gym level to use available slots info for maps without gym-info

and probably some more things

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pogom/models.py [hunk 1]
```diff
@@ -2323,48 +2321,45 @@ def parse_map(args, map_dict, step_location, db_update_queue, wh_update_queue,
...
+                        raid_pokemon = raid_info.get('raid_pokemon', {})
+                        if raid_pokemon:
+                            raids[f['id']].update({
+                                'pokemon_id': raid_pokemon['pokemon_id'],
+                                'cp': raid_pokemon['cp'],
+                                'move_1': raid_pokemon['move_1'],
+                                'move_2': raid_pokemon['move_2']
+                            })
+
+                        if args.webhooks:
```
Comment (FrostTheFox) [inline pogom/models.py]: Shouldn't this still have `and not args.webhook_updates_only:`?

**Comment by FrostTheFox** [comment]
@Arengo Please review when possible

File: pogom/models.py (hunk 1 above)
Comment (friscoMad) [inline pogom/models.py]: I am not sure as wH updates only has always been really cryptic to me so I am not really sure what should it do.
But that reminds me that we haven't added raids to WH cache

File: pogom/models.py [hunk 2]
```diff
@@ -2323,48 +2321,45 @@ def parse_map(args, map_dict, step_location, db_update_queue, wh_update_queue,
...
-                            'move_1': raid_pokemon['move_1'],
-                            'move_2': raid_pokemon['move_2']
-                        })
-
-                    if args.webhooks and not args.webhook_updates_only:
-                        wh_update_queue.put(('raid', {
-                            'gym_id': b64encode(str(raids[f['id']]['gym_id'])),
-                            'latitude': f['latitude'],
-                            'longitude': f['longitude'],
+                # Only stops and gyms right now.
```
Comment (Arengo) [inline pogom/models.py]: Unneeded comment (in general).

File: pogom/models.py (hunk 1 above)
Comment (Arengo) [inline pogom/models.py]: Don't understand webhook updates very well, too. Afaik, writing the TeleRaid notification, I only get webhooks from gyms with a new ``last_changed`` it seems.

**Comment by FrostTheFox** [comment]
@arengo, RE: gyms with different last_changes, iirc the webhook queue is setup such that it will only send an object once until it is updated

**Comment by friscoMad** [comment]
Should be fixed now

File: pogom/models.py (hunk 1 above)
Comment (FrostTheFox) [inline pogom/models.py]: --webhook-updates-only
                    Only send updates (Pokemon & lured pokestops). [env
                    var: POGOMAP_WEBHOOK_UPDATES_ONLY]

Comment (FrostTheFox) [inline pogom/models.py]: This is not a pokemon or lured stop, so it should be `and not args.webhook_updates_only`

**Comment by FrostTheFox** [comment]
Fixed the args.webhook_updates_only.

File: pogom/models.py [hunk 3]
```diff
@@ -2323,48 +2323,44 @@ def parse_map(args, map_dict, step_location, db_update_queue, wh_update_queue,
...
+                            raids[f['id']].update({
+                                'pokemon_id': raid_pokemon['pokemon_id'],
+                                'cp': raid_pokemon['cp'],
+                                'move_1': raid_pokemon['move_1'],
+                                'move_2': raid_pokemon['move_2']
+                            })
+
+                        if args.webhooks and not args.webhook_updates_only:
+                            wh_raid = raids[f['id']].copy()
+                            wh_raid.update({
```
Comment (Arengo) [inline pogom/models.py]:
```python
'pokemon_id': raid_pokemon['pokemon_id'],
'cp': raid_pokemon['cp'],
'move_1': raid_pokemon['move_1'],
'move_2': raid_pokemon['move_2']
```

Is missing for the webhook!

**Comment by friscoMad** [comment]
Not missing, it clones the whole object and only updates what's
needed, lat/long and times that need to be unix time instead of date
objects.


2017-06-30 9:24 GMT+02:00 Arengo <notifications@github.com>:

File: pogom/models.py (hunk 3 above)
Comment (PKMTracker) [inline pogom/models.py]: Can confirm, the above never get sent over the webhook. I've MITM all the data, and this is as detailed as it gets:

{"message": {
    "spawn": 1498824099618, 
    "gym_id": "MDQxNDRiZmM2YzNlNDY2YWE2ZThmYTg2ZGM0MTM1MTMuMTY=", 
    "battle": 1498831299618, 
    "cp": 0, 
    "move_1": 0, 
    "move_2": 0, 
    "end": 1498834899618, 
    "level": 2, 
    "pokemon_id": 0, 
    "longitude": -114.071718, 
    "latitude": 51.080624
    }, 
"type": "raid"
}

Hope it helps!

**Pull request merged**